accounts-db/src/ancient_append_vecs.rs

```diff
@@ -177,8 +179,10 @@ impl AncientSlotInfos {
                     * tuning.percent_of_alive_shrunk_data
                     / 100,
             );
+        self.best_slots_to_shrink = Vec::with_capacity(self.shrink_indexes.len());
         for info_index in &self.shrink_indexes {
             let info = &mut self.all_infos[*info_index];
+            self.best_slots_to_shrink.push((info.slot, info.capacity));
```

Comment:
Should we only insert the info for the ancient storage after cut-off? 
Those ancient storages before the cut-off may have already be shrunk by 'shrink_ancient_slots()' before we call 'shrink_candidate_slots?

https://github.com/anza-xyz/agave/blob/18059246e5e845855a2d4f8f0441a4fabd101ca6/runtime/src/accounts_background_service.rs#L371-L375

Reply:
i.e. 
```
@@ -1,11 +1,11 @@
        for info_index in &self.shrink_indexes {
             let info = &mut self.all_infos[*info_index];
-            self.best_slots_to_shrink.push((info.slot, info.capacity));
             if bytes_to_shrink_due_to_ratio.0 >= threshold_bytes {
                 // we exceeded the amount to shrink due to alive ratio, so don't shrink this one just due to 'should_shrink'
                 // It MAY be shrunk based on total capacity still.
                 // Mark it as false for 'should_shrink' so it gets evaluated solely based on # of files.
                 info.should_shrink = false;
+                self.best_slots_to_shrink.push((info.slot, info.capacity));
             } else {
                 bytes_to_shrink_due_to_ratio += info.alive_bytes;
             }
```